src/main/java/de/tum/cit/aet/artemis/exam/service/ExamDateService.java

```diff
@@ -99,7 +110,17 @@ public boolean isIndividualExerciseWorkingPeriodOver(Exam exam, StudentParticipa
             return false;
         }
 
-        var optionalStudentExam = studentExamRepository.findByExamIdAndUserId(exam.getId(), studentParticipation.getParticipant().getId());
+        Optional<StudentExam> optionalStudentExam;
+        // There are multiple test exams possible for one exam, which implies that there are multiple student exams for one exam.
+        // For test exams we try to find the latest student exam
+        // For real exams we try to find the only existing student exam
+        if (exam.isTestExam()) {
+            optionalStudentExam = studentExamRepository.findFirstByExamIdAndUserIdOrderByIdDesc(exam.getId(), studentParticipation.getParticipant().getId());
+        }
+        else {
+            optionalStudentExam = studentExamRepository.findByExamIdAndUserId(exam.getId(), studentParticipation.getParticipant().getId());
+        }
```

Comment:
```suggestion
        optionalStudentExam = studentExamRepository.findFirstByExamIdAndUserIdOrderByIdDesc(exam.getId(), studentParticipation.getParticipant().getId());
```
Shouldn't this work in both scenarios?

Reply:
fixed